trulens_eval/trulens_eval/feedback/provider/base.py

```diff
@@ -17,3 +25,920 @@ def __init__(self, name: str = None, **kwargs):
+
+        Args:
+            system_prompt (str): A pre-formated system prompt
+
+        Returns:
+            The score and reason metadata if available.
+        """
+        llm_messages = [{"role": "system", "content": system_prompt}]
+        if user_prompt is not None:
+            llm_messages.append({"role": "user", "content": user_prompt})
+
+        response = self.endpoint.run_me(
+            lambda: self._create_chat_completion(
+                model=self.model_engine, temperature=0.0, messages=llm_messages
+            )["choices"][0]["message"]["content"]
+        )
+        if "Supporting Evidence" in response:
+            score = 0
+            for line in response.split('\n'):
+                if "Score" in line:
+                    score = re_1_10_rating(line) / normalize
+            return score, {"reason": response}
+        else:
+            return re_1_10_rating(response) / normalize
+
+    def qs_relevance(self, question: str, statement: str) -> float:
+        """
+        Uses chat completion model. A function that completes a
+        template to check the relevance of the statement to the question.
+
+        feedback = Feedback(provider.qs_relevance).on_input_output() 
+        ```
+        The `on_input_output()` selector can be changed. See [Feedback Function Guide](https://www.trulens.org/trulens_eval/feedback_function_guide/)
+        
+        Usage on RAG Contexts:
+        ```
+        feedback = Feedback(provider.qs_relevance).on_input().on(
+            TruLlama.select_source_nodes().node.text # See note below
+        ).aggregate(np.mean) 
+
+        ```
+        The `on(...)` selector can be changed. See [Feedback Function Guide : Selectors](https://www.trulens.org/trulens_eval/feedback_function_guide/#selector-details)
+
+
+        Args:
+            question (str): A question being asked. 
+            statement (str): A statement to the question.
+
+        Returns:
+            float: A value between 0 and 1. 0 being "not relevant" and 1 being "relevant".
+        """
+        return re_1_10_rating(
+            self.endpoint.run_me(lambda:
+                self._create_chat_completion(
+                    prompt=
+                                str.format(
+                                    prompts.QS_RELEVANCE,
+                                    question=question,
+                                    statement=statement
+                                )
+                )
+            ) / 10
+        )
+    def qs_relevance_with_cot_reasons(
+        self, question: str, statement: str
+    ) -> float:
+        """
+        Uses chat completion model. A function that completes a
+        template to check the relevance of the statement to the question.
+        Also uses chain of thought methodology and emits the reasons.
+
+        **Usage:**
+        ```
+        feedback = Feedback(provider.qs_relevance_with_cot_reasons).on_input_output() 
```

Comment:
maybe for each of these add

Intantiate a provider with either [OpenAI](https://www.trulens.org/trulens_eval/api/feedback/#trulens_eval.trulens_eval.feedback.provider.openai.OpenAI) or [BedRock](https://www.trulens.org/trulens_eval/api/feedback/#trulens_eval.trulens_eval.feedback.provider.bedrock.Bedrock)

Reply:
Added new provider api reference pages that should make this more clear.